packages/block-library/src/navigation/style.scss

```diff
@@ -169,3 +169,71 @@
 .items-justified-right > ul {
 	justify-content: flex-end;
 }
+
+.wp-block-navigation-placeholder__actions {
+	display: flex;
+	align-items: flex-start;
+
+	.components-base-control {
+		margin-right: 5px;
+	}
+
+	.components-select-control__input {
+		margin: 0;
+		min-height: 36px; // match button height
+	}
+
+	.wp-block-navigation-placeholder__button {
+		height: 36px;
+	}
+
+
+}
+
+.wp-block-navigation-placeholder {
+
+	.components-custom-select-control__button {
+		height: auto; // overide default button styles
+		margin-bottom: 0; // cancel bottom margin
+		min-width: 220px; // avoids control size jump depending on selection
+
+		// Specificity overide
+		&.components-custom-select-control__button {
+			padding-right: 24px; // provide sufficient space for dropdown arrow
+		}
+	}
+
+	.components-custom-select-control__item {
+		cursor: pointer;
+	}
```

Comment:
Again, it feels like this should be part of the component.

Reply:
I had assumed this was just an anomaly due to my overwriting some of the styles here, but testing in Storybook I can see it's an issue there as well.

https://wordpress.github.io/gutenberg/?path=/story/components-customselectcontrol--default

I might remove from this PR and make an Issue to fix separately.

Reply:
@talldan here is a PR for that https://github.com/WordPress/gutenberg/pull/22926

Reply:
PR merged and code removed from this PR.